lib/Console/Command/Parse.php

```diff
@@ -153,6 +130,48 @@ protected function execute(InputInterface $input, OutputInterface $output) {
         }
     }
 
+    /**
+     * Fetch commits from mesa's git.
+     *
+     * @param string $gl3Path The path of the GL3.txt file.
+     * @param OutputInterface $output The output to write to.
+     */
+    protected function fetchCommits($gl3Path, OutputInterface $output) {
+        $gitLog = new \Mesamatrix\Git\ProcessBuilder(array(
+            'log', '--pretty=format:%H|%at|%aN|%cN|%ct|%s', '--reverse',
+            \Mesamatrix::$config->getValue('git', 'oldest_commit').'..', '--',
+            $gl3Path
+        ));
+        $proc = $gitLog->getProcess();
+        $this->getHelper('process')->mustRun($output, $proc);
+
+        $commitLines = explode(PHP_EOL, $proc->getOutput());
+        if (empty($commitLines)) {
+            // No commit? There must be a problem.
+            \Mesamatrix::$logger->error("No commit found.");
+            return NULL;
+        }
+
+        // Create commit list.
+        $commits = array();
+        foreach ($commitLines as $commitLine) {
+            $commitData = array();
+            if (preg_match('/(.+)\|(.+)\|(.+)\|(.+)\|(.+)\|(.+)/',
```

Comment:
I preferred the `explode()` version here - it is more expressive than a regex, and easier to read :smile: